core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -236,12 +253,67 @@ public List<String> getTabletServers() {
   }
 
   @Override
+  public Set<Server> getServers(ServerType type) {
```

Comment:
I see this is where the type enum is used by users. But, you could just as easily get rid of the enum, and users can pass a `Predicate<Server>`, which is far more flexible, and they can do things like `getServers(s -> s.getHost().startsWith("somePrefix"));` or `getServers(s -> s instanceof TabletServer)`.

Reply:
If I use the Predicate then I have to test against every single server each time this is called, right? This is replacing `getTabletServers`, `getScanServers` and `getManagerLocations`. If the user want's everything, then I think it makes sense that they call this method N times instead of us gathering all of the information every time.

Reply:
Ok, I think I figured out a solution for this.

Reply:
Solution implemented in f63dce0